src/adaptors/tcp/tcp_adaptor.c

```diff
@@ -1209,9 +1250,14 @@ static qdr_tcp_connection_t *qdr_tcp_connection(qd_tcp_listener_t *listener, qd_
     assert(tcp_stats);
     assert(server);
 
+    if (tc->config->adaptor_config->idle_timeout) {
```

Comment:
Move this into the above if (listener) so we can have idle timers only for connections on the listener side ?

Reply:
We may get a request to add timeout support for the connector side as well. Still not carved in stone but if so I'd rather not conventionalize the code on the type of configuration. We'll just have to revert that change if so.